runtime/src/bank/fee_distribution.rs

```diff
@@ -70,26 +83,32 @@ impl Bank {
             fee_budget_limits.prioritization_fee,
             FeeFeatures::from(self.feature_set.as_ref()),
         );
-        let (reward, _burn) =
-            self.calculate_reward_and_burn_fee_details(&CollectorFeeDetails::from(fee_details));
+        let BurnedTransactionFee {
+            deposit: reward,
+            burn: _,
+        } = self.calculate_reward_and_burn_fee_details(&CollectorFeeDetails::from(fee_details));
         reward
     }
 
     fn calculate_reward_and_burn_fee_details(
         &self,
         fee_details: &CollectorFeeDetails,
-    ) -> (u64, u64) {
-        let (deposit, burn) = if fee_details.transaction_fee != 0 {
-            self.burn_transaction_fee(fee_details.transaction_fee)
-        } else {
-            (0, 0)
-        };
-        (deposit.saturating_add(fee_details.priority_fee), burn)
+    ) -> BurnedTransactionFee {
+        if fee_details.transaction_fee == 0 {
+            return BurnedTransactionFee::default();
+        }
+
+        let mut burned_fee = self.burn_transaction_fee(fee_details.transaction_fee);
+        burned_fee.add_deposit(fee_details.priority_fee);
+        burned_fee
     }
 
-    fn burn_transaction_fee(&self, fees: u64) -> (u64, u64) {
-        let burned = fees * self.burn_percent() / 100;
-        (fees - burned, burned)
+    fn burn_transaction_fee(&self, fees: u64) -> BurnedTransactionFee {
```

Comment:
can we prefix with `calculate`? this isn't actually burning the fee, it's calculating what should be burned.

Reply:
or actually - I think we should delete this function, and just make it part of `calculate_reward_and_burn_fee_details`. The `add_deposit` function can go away with this change

Reply:
Like the idea. bee69840a27782f4a7a49c07935bda4023eadc35 does it, also renames struct, removes `add_deposit`.